### PR TITLE
Add leavesMap and fix broken tests on `identity-vanchor` and `vanchorForest`

### PR DESCRIPTION
**Pull request by semaraugusto**
* Fixing tests for identity-vanchor and vanchor-forest
* Add leavesMap parameter to transaction in both classes

### CODE REVIEW

**Comment by drewstone** [comment]
@shekohex possibly this PR fixed some issues you're having. We should review and merge asap. CC @saiakilesh 

There does seem to be a lot of commented out code here still @semaraugusto. Would be good to clean it up and make this PR ready to merge. It would also be good to write some cross chain tests for these anchors to ensure the changes made here are actually working for the intended functionality.

**Comment by drewstone** [comment]
Merge conflicts here still, any update?

**Comment by semaraugusto** [comment]
I'm finishing the cross-chain tests for the identity-vanchor. One can be part of a semaphore group on chain A and prove membership (which allows him to transact) on chainB. The cross-chain withdraw test is still failing. I'll do a clean up and merge when I get it working

**Review by saiakilesh — APPROVED** [review]

File: packages/contracts/test/identityVAnchor/identityVAnchor.test.ts [hunk 1]
```diff
@@ -1488,30 +1519,326 @@ describe.skip('IdentityVAnchor for 2 max edges', () => {
...
+      await token.mint(john.address, BigNumber.from(1e10).toString());
+      const johnBalanceBeforeDeposit = await token.balanceOf(john.address);
+      const txOptions: TransactionOptions = {
+        keypair: johnKeypair,
+        externalLeaves: ganacheGroup.members.map((bignum: BigNumber) =>
+          hexToU8a(bignum.toHexString())
+        ),
+      };
+      let johnLeaf = johnKeypair.getPubKey();
+      // const updateEdgeTx = await semaphore.updateEdge(groupId.toNumber(), ganacheGroup.root.toString(), 1, chainID2)
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identityVAnchor.test.ts:1750]: Are these comments needed?

File: packages/contracts/test/identityVAnchor/identityVAnchor.test.ts [hunk 2]
```diff
@@ -1488,30 +1519,326 @@ describe.skip('IdentityVAnchor for 2 max edges', () => {
...
+      //
+      // const updateEdgeReceipt = await updateEdgeTx.wait()
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identityVAnchor.test.ts:1752]: Here too?

File: packages/contracts/test/identityVAnchor/identityVAnchor.test.ts [hunk 3]
```diff
@@ -1488,30 +1519,326 @@ describe.skip('IdentityVAnchor for 2 max edges', () => {
...
+        groupId.toNumber(),
+        group.root.toString(),
+        2,
+        chainID
+      );
+
+      // update group edges cross-chain
+      ganacheGroup.updateEdge(chainID, group.root.toString());
+
+      // let john_addmember_tx = await ganacheSemaphore.contract
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identityVAnchor.test.ts:1648]: Is this needed?

File: packages/contracts/test/identityVAnchor/identityVAnchor.test.ts [hunk 4]
```diff
@@ -1488,30 +1519,326 @@ describe.skip('IdentityVAnchor for 2 max edges', () => {
...
+      ganacheGroup.addMember(johnLeaf);
+      let john_addmember_tx = await ganacheSemaphore.contract
+        .connect(ganacheWallet)
+        .addMember(groupId, johnLeaf, { gasLimit: '0x5B8D80' });
+      // const receipt = await alice_addmember_tx.wait();
+      expect(john_addmember_tx)
+        .to.emit(ganacheSemaphore.contract, 'MemberAdded')
+        .withArgs(groupId, johnLeaf, ganacheGroup.root);
+
+      // const tx2 = await ganacheSemaphore.addMember(groupId.toNumber(), group.root.toString(), 2, chainID);
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identityVAnchor.test.ts:1637]: Can this be removed?

File: packages/contracts/test/identityVAnchor/identityVAnchor.test.ts [hunk 5]
```diff
@@ -1488,30 +1519,326 @@ describe.skip('IdentityVAnchor for 2 max edges', () => {
...
+        ganacheWallet.address,
+        maxEdges
+      );
+      let johnLeaf = johnKeypair.getPubKey();
+      ganacheGroup = new LinkedGroup(levels, maxEdges, BigInt(defaultRoot));
+      ganacheGroup.addMember(johnLeaf);
+      let john_addmember_tx = await ganacheSemaphore.contract
+        .connect(ganacheWallet)
+        .addMember(groupId, johnLeaf, { gasLimit: '0x5B8D80' });
+      // const receipt = await alice_addmember_tx.wait();
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identityVAnchor.test.ts:1632]: Can this be removed?

File: packages/anchors/src/VAnchorForest.ts [hunk 6]
```diff
@@ -384,56 +398,49 @@ export class VAnchorForest extends WebbBridge {
...
+        );
+      }
+      const treeElements: Uint8Array[] = leavesMap[treeChainId];
+      const forestElements: Uint8Array[] | undefined = txOptions.externalLeaves;
+      if (forestElements === undefined) {
+        throw new Error(
+          'Need to specify forestElements on txOptions in order to generate merkleProof correctly'
+        );
+      }
+      // const forestElements = leavesMap[treeChainId]
```
Comment (drewstone) [inline packages/anchors/src/VAnchorForest.ts:439]: Can this be removed?

File: packages/anchors/src/Common.ts [hunk 7]
```diff
@@ -282,4 +299,153 @@ export class WebbBridge {
...
+    if (isOpenVAnchorContract(this.contract)) {
+      throw new Error('OpenVAnchor contract does not support the `transact` method');
+    }
+    // Validate input utxos have a valid originChainId
+    // this.validateInputs(inputs);
+
+    const [overrides, txOptions] = splitTransactionOptions(overridesTransaction);
+
+    // Default UTXO chain ID will match with the configured signer's chain ID
+    // inputs = await this.padUtxos(inputs, 16);
```
Comment (drewstone) [inline packages/anchors/src/Common.ts:400]: Can this be removed?

File: packages/anchors/src/Common.ts [hunk 8]
```diff
@@ -282,4 +299,153 @@ export class WebbBridge {
...
+    relayer: string,
+    wrapUnwrapToken: string,
+    leavesMap: Record<string, Uint8Array[]>, // subtree
+    overridesTransaction?: OverridesWithFrom<PayableOverrides> & TransactionOptions
+  ): Promise<ethers.ContractReceipt> {
+    if (isOpenVAnchorContract(this.contract)) {
+      throw new Error('OpenVAnchor contract does not support the `transact` method');
+    }
+    // Validate input utxos have a valid originChainId
+    // this.validateInputs(inputs);
```
Comment (drewstone) [inline packages/anchors/src/Common.ts:395]: Can this be removed?

**Review by drewstone — CHANGES_REQUESTED** [review]
LG, @semaraugusto @saiakilesh can we make sure in future reviews that you guys are also checking for code style and documentation? This PR took a ~week and there's still commented code that is left here and not a meaningful improvement of documentation in the code. We can do better here

**Pull request merged**